src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1436,7 +1436,7 @@ public String create(final String path, byte data[], List<ACL> acl,
         request.setData(data);
         request.setFlags(createMode.toFlag());
         request.setPath(serverPath);
-        if (acl != null && acl.size() == 0) {
+        if (acl == null || acl.size() == 0) {
```

Comment:
Wouldn't be better to account for `null` entries in the `acl` List like below?

```
if (acl == null || acl.size() == 0 || acl.contains(null)) {
```

The same applies to other changes in this file.